index.js

```diff
@@ -0,0 +1,314 @@
+/**
+ * Full fhirconvert function: validates, converts, then identifies variables.
+ * @param {string} str - inputted normal syntax expression
+ * @param {Array} vars - array of usable variables entered by user
+ * @returns converted fhirpath expression
+ */
+export function fhirconvert(str, vars) {
+  if (validate(str, vars)) {
+    return varfind(convert(str), vars);
+  } else {
+    return null;
+  }
+}
+
+/**
+ * Verifies normal syntax by confirming var names, function names,
+ * and number of parenthesis.
+ * @param {string} str - inputted normal syntax expression
+ * @param {Array} vars - array of usable variables
+ * @returns boolean, valid or invalidh
+ */
+export function validate(str, vars) {
+  let funs = [
+    "CEILING",
+    "FLOOR",
+    "ABS",
+    "TRUNCATE",
+    "EXP",
+    "SQRT",
+    "LN",
+    "LOG",
+    "ceiling",
+    "floor",
+    "abs",
+    "truncate",
+    "exp",
+    "sqrt",
+    "ln",
+    "log",
+    "and"
+  ];
+  var lcount = 0;
+  var rcount = 0;
+  var len = str.length;
+  var substr = "";
+  for (var i = 0; i < len; i++) {
+    if (str[i] == "(") {
+      lcount += 1;
+    }
+    if (str[i] == ")") {
+      rcount += 1;
+    }
+    if (/[a-zA-Z]/.test(str[i])) {
+      substr = substr + str[i];
+    }
+    if (str[i + 1] == null || !/[a-zA-Z]/.test(str[i + 1])) {
+      if (funs.includes(substr) || vars.includes(substr) || substr == "") {
+        substr = "";
+      } else {
+        return false;
+      }
+    }
+  }
+  return lcount == rcount;
+}
+
+/**
+ * Identifies convertable functions in expression and converts them recursively.
+ * @param {string} str - inputted normal syntax expression
+ * @returns expression with converted functions
+ */
+export function convert(str) {
+  let funs = [
+    "CEILING",
+    "FLOOR",
+    "ABS",
+    "TRUNCATE",
+    "EXP",
+    "SQRT",
+    "LN",
+    "ceiling",
+    "floor",
+    "abs",
+    "truncate",
+    "exp",
+    "sqrt",
+    "ln",
+  ];
+  var count = 0;
+  if (str.includes("^")) {
```

Comment:
Another common syntax is **.  That is JavaScript.  Can you allow both?

Reply:
Fixed